tasks/connectors/edgescan/lib/edgescan_vulnerability.rb

```diff
@@ -49,12 +45,33 @@ def to_kenna_finding
           }
         end
 
-        def to_corresponding_kenna_asset
-          { "external_id" => external_asset_id, "tags" => asset.tags }
+        def to_kenna_asset
+          base = {
+            "external_id" => external_asset_id,
+            "tags" => asset["tags"],
+            "application" => asset_application_id
+          }
+          if ip_address?
+            base["ip"] = location
+          elsif url?
+            base["url"] = location
+          else
+            base["hostname"] = location
+          end
+          if host
+            base["ip"] = host["location"]
+            base["hostname"] = host["hostnames"]&.first
+            base["os"] = host["os_name"]
```

Comment:
`os` is intended to mark the OS type not the OS version - to specify the OS version, use the `os_version` property: https://help.kennasecurity.com/hc/en-us/articles/360026413111-Kenna-Data-Importer-JSON-Connector-